### ISSUE TITLE
Azimuthal range issue in integrate2d_legacy  when discontinuity is set to zero.

### ISSUE THREAD

**Issue by kif** [silx-kit/pyFAI#1533]
Reported by Thomas Z.

**Comment by kif** [comment]
```
ai.setChiDiscAtZero()
jupyter.plot2d(ai.integrate2d(img, 500, 360, dark=dark, mask=mask, unit="r_mm"))``
![image](https://user-images.githubusercontent.com/1018880/124884362-7e4c7c00-dfd2-11eb-90aa-5a6cf6fd1651.png)

**Comment by kif** [comment]
Enforcing  `azimuth_range=(0,360)` fixes the issue.

**Comment by kif** [comment]
Probably a bug in the definition of the default azimuthal range, wrongly interpreted in some of the integration engines themselves.

**Comment by kif** [comment]
wont't be fixed since the method is deprecated in 0.21
